### ISSUE TITLE
Dynamically Replacing YAxis throws Highcharts error

### ISSUE THREAD

**Issue by cbergmiller** [whawker/react-jsx-highcharts#21]
First let me say that i think that **react-jsx-highcharts** is an excellent approach for integrating highcharts in react apps! I would like to try and use it for the projects i am responsible for at work.

I ran into an issue when i tried to dynamically replace y-axis:
```jsx
<HighchartsChart>
    <Chart/>
    <Title>{title}</Title>
    <Legend layout="vertical" align="right" verticalAlign="middle"/>
    <XAxis type="datetime">
        <XAxis.Title>Time</XAxis.Title>
    </XAxis>
    {selectedTypes.map(type =>
        <YAxis id={`y-axis-${type.id}`}
               key={type.id}>
            <YAxis.Title>{`${type.name} [${type.unit}]`}</YAxis.Title>
            <LineSeries id={`series-${type.id}`}
                          name={type.name}
                          data={this.getSeriesDataByTypeId(type.id)}/>
        </YAxis>
    )}
</HighchartsChart>
```
This works fine until one of the `selectedTypes` is removed. The following error is thrown:
```
highcharts.js:336 Uncaught TypeError: Cannot read property 'options' of undefined
    at L.update (highcharts.js:336)
    at react-jsx-highcharts.js:1021
    at Object.update (react-jsx-highcharts.js:1067)
    at AxisTitle.updateAxisTitle (react-jsx-highcharts.js:3335)
    at AxisTitle.componentWillUnmount (react-jsx-highcharts.js:3324)
    at ReactCompositeComponent.js:408
    at measureLifeCyclePerf (ReactCompositeComponent.js:75)
    at ReactCompositeComponentWrapper.unmountComponent (ReactCompositeComponent.js:407)
    at Object.unmountComponent (ReactReconciler.js:78)
    at ReactCompositeComponentWrapper.unmountComponent (ReactCompositeComponent.js:417)
```
Is removing y-axis not supported or am i doing something wrong?

**Comment by cbergmiller** [comment]
I prepared a CodeSandbox to demonstrate the issue:
https://codesandbox.io/s/647xv8plpr

On first click on the button it throws an uncaught TypeError: `this.props.remove is not a function`

**Comment by whawker** [comment]
Ok, so it seems that when your remove an Axis, Highcharts also removes the series bound to that axis, makes sense of course.

But it means that React JSX Highcharts was calling remove on a Series that had already been cleaned up. The commit above should fix that.

**Comment by whawker** [comment]
Just published `react-jsx-highcharts@1.3.4-pre.1` as a beta. Let me know if that works for you. (I haven't published `react-jsx-highstock`, but if you need me to, let me know).

**Comment by whawker** [comment]
P.p.s that CodeSandbox example is using a really old version of `react-jsx-highcharts` by the way.
